packages/foam-core/src/core.ts

```diff
@@ -0,0 +1,89 @@
+import { Graph, Edge } from 'graphlib'
+
+
+type ID = string
+
+export interface Link {
+  from: ID
+  to: ID
+  text: string
+}
+
+export interface NoteLink {
+  to: ID
+  text: string
+}
+
+
+export class Note {
+  public id: ID
+  public title: string
+  public source: string
+  public path: string
+  public links: NoteLink[]
+
+  constructor(id: ID, title: string, links: NoteLink[], path: string, source: string) {
```

Comment:
maybe use Partials here? https://stackoverflow.com/a/37682352

Reply:
I was wondering whether this needs to be a class in the first place, or just a type/interface - what do you think?